### PR TITLE
Unify nas searchspace

### PR DESCRIPTION
**Pull request by Crysple**
- Unify NAS search space with `_type` -- "mutable_type"
- Add ut
- Update random search tuner

### CODE REVIEW

**Comment by QuanluZhang** [comment]
@Crysple great job! please update the search space spec doc accordingly

File: examples/tuners/random_nas_tuner/random_nas_tuner.py [hunk 1]
```diff
@@ -7,7 +7,9 @@ def random_archi_generator(nas_ss, random_state):
     '''
     chosen_archi = {}
     print("zql: nas search space: ", nas_ss)
-    for block_name, block in nas_ss.items():
+    for block_name, block_value in nas_ss.items():
+        assert block_value['_type'] == "mutable_layer", "Random NAS Tuner only receives NAS search space"
```
Comment (xuehui1991) [inline examples/tuners/random_nas_tuner/random_nas_tuner.py]: why not point out we need "mutable_layer" key here.

**Comment by Crysple** [comment]
> @Crysple great job! please update the search space spec doc accordingly

Have added. Please help review

File: examples/tuners/random_nas_tuner/random_nas_tuner.py (hunk 1 above)
Comment (Crysple) [reply examples/tuners/random_nas_tuner/random_nas_tuner.py]: I have modify the info to explicitly pointed this out

**Pull request merged**